File: March2021/SwappingNodesinaLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    // TC : O(n)
    // SC : O(1)
    public ListNode swapNodes(ListNode head, int k) {
        k = k-1;
        ListNode firstN = head;
        while(k>0){
            firstN = firstN.next;
            k--;
        }

        ListNode f = firstN;
        ListNode secondN = head;
        while(f.next!=null){
            secondN = secondN.next;
            f = f.next;
        }

        int temp = firstN.val;
        firstN.val = secondN.val;
        secondN.val = temp;

        return head;


    }
}

File: April2022/Kotlin/SwappingNodesInALinkedList.kt
// Author: Shobhit Behl (LC: shobhitbruh)
var len = 1
        var j = head
        while (j != null) {
            len++
            j = j.next
        }
        var i = head
        var p = 1
        while (p < k) {
            i = i!!.next
            p++
        }
        p = 1
        var d = head
        while (p < len - k) {
            d = d!!.next
            p++
        }
        val temp = d!!.`val`
        d.`val` = i!!.`val`
        i.`val` = temp
        return head